Modules/_csv.c

```diff
@@ -1379,6 +1406,9 @@ csv_writer(PyObject *module, PyObject *args, PyObject *keyword_args)
     self->rec_len = 0;
     self->num_fields = 0;
 
+    self->error_obj = module_state->error_obj;
+    Py_INCREF(self->error_obj);
```

Comment:
nitpick, but you can leave the code as it it.

```suggestion
    self->error_obj = Py_NewRef(module_state->error_obj);    
```